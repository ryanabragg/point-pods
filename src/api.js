import localforage from 'localforage';
import React from 'react';

const api = {
  store: localforage.createInstance({
    name: 'Point Pods',
    storeName: 'point_pods',
    description: 'some description'
  }),
};

export default api;
export const APIContext = React.createContext(api);
export function withAPI(Component) {
  return function APIComponent(props) {
    return (
      <APIContext.Consumer>
        {context => <Component {...props} api={context} />}
      </APIContext.Consumer>
    );
  };
}
