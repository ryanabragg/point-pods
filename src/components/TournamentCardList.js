import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';
import { withAPI } from '../api';

import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';

const styles = theme => ({
  root: {
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'space-evenly',
    alignItems: 'flex-start',
    alignContent: 'flex-start',
  },
  card: {
    minWidth: 275,
  },
  right: {
    marginLeft: 'auto',
  },
});

class TournamentCardList extends Component {
  state = {
    tournaments: [],
  };

  componentDidMount() {
    this.props.api.Tournaments.all()
      .then(list => this.setState({ tournaments: list }))
      .catch(error => console.log(error.message));
  }

  render() {
    const { classes } = this.props;
    const { tournaments } = this.state;
    const cards = this.props.filterByDone
      ? tournaments.filter(t => t.done === this.props.done)
      : tournaments;
    return (
      <div className={classes.root}>
        {cards.map(({id, name, type, players}) => (
          <Card key={id} className={classes.card}>
            <CardHeader
              title={name}
              subheader={type}
            />
            <CardContent>
              <Typography>
                {`${players.length} Total Players`}
              </Typography>
              <Typography>
                {`${players.filter(t => !t.dropped).length} Active Players`}
              </Typography>
            </CardContent>
            <CardActions>
              <Button size='small'
                className={classes.right}
                href={`/tournament/${id}`}
              >
                Details
              </Button>
            </CardActions>
          </Card>
        ))}
      </div>
    );
  }
}

TournamentCardList.defaultProps = {
  filterByDone: false,
  done: false,
};

TournamentCardList.propTypes = {
  api: PropTypes.object.isRequired, // added by withAPI
  classes: PropTypes.object.isRequired, // added by withStyles
  filterByDone: PropTypes.bool,
  done: PropTypes.bool,
};

export default withStyles(styles, { withTheme: true })(withAPI(TournamentCardList));
