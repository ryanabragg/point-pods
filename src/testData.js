import api from './api';

const testSettings = {
  pairingMethod: api.pairingMethods.HISTORIC,
  pairingMethodInitial: api.pairingMethods.HISTORIC,
  podSizeMinimum: 4,
  podSizeMaximum: 7,
};

const testPlayers = [{
  id: 'mock0',
  name: 'test0',
  points: 0,
}, {
  id: 'mock1',
  name: 'test1',
  points: 1,
}, {
  id: 'mock2',
  name: 'test2',
  points: 2,
}, {
  id: 'mock3',
  name: 'test3',
  points: 3,
}, {
  id: 'mock4',
  name: 'test4',
  points: 4,
}, {
  id: 'mock5',
  name: 'test5',
  points: 5,
}, {
  id: 'mock6',
  name: 'test6',
  points: 6,
}, {
  id: 'mock7',
  name: 'test7',
  points: 7,
}, {
  id: 'mock8',
  name: 'test8',
  points: 8,
}, {
  id: 'mock9',
  name: 'test9',
  points: 9,
}];

const testTournaments = [{
  id: 'mock0',
  name: 'test0',
  category: 'cat0',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris vitae tellus nunc. Etiam urna tortor, laoreet vitae commodo porttitor, mattis id purus.',
  date: new Date('2018-01-01'),
  done: false,
  staging: true,
  pairingMethod: api.pairingMethods.RANDOM,
  pairingMethodInitial: api.pairingMethods.RANDOM,
  podSizeMinimum: 3,
  podSizeMaximum: 4,
  rounds: 1,
  players: [{
    id: 'mock0',
    name: 'test0',
    points: 3,
    participated: true,
    dropped: true,
    1: { pod: 1, points: 3 },
  }, {
    id: 'mock1',
    name: 'test1',
    points: 1,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 1 },
  }, {
    id: 'mock2',
    name: 'test2',
    points: 2,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 2 },
  }, {
    id: 'mock3',
    name: 'test3',
    points: 0,
    participated: false,
    dropped: false,
  }],
}, {
  id: 'mock1',
  name: 'test1',
  category: 'cat1',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed molestie suscipit suscipit. Nullam sollicitudin tempus pretium.',
  date: new Date('2018-02-01'),
  done: true,
  staging: false,
  pairingMethod: api.pairingMethods.POINTS,
  pairingMethodInitial: api.pairingMethods.RANDOM,
  podSizeMinimum: 3,
  podSizeMaximum: 5,
  rounds: 2,
  players: [{
    id: 'mock0',
    name: 'test0',
    points: 0,
    participated: true,
    dropped: true,
    1: { pod: 1, points: 0 },
    2: { pod: 1, points: 0 },
  }, {
    id: 'mock1',
    name: 'test1',
    points: 1,
    participated: true,
    dropped: true,
    1: { pod: 1, points: 1 },
    2: { pod: 2, points: 1 },
  }, {
    id: 'mock2',
    name: 'test2',
    points: 2,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 2 },
    2: { pod: 1, points: 2 },
  }, {
    id: 'mock3',
    name: 'test3',
    points: 3,
    participated: true,
    dropped: false,
    1: { pod: 2, points: 3 },
    2: { pod: 2, points: 3 },
  }, {
    id: 'mock4',
    name: 'test4',
    points: 4,
    participated: true,
    dropped: false,
    1: { pod: 2, points: 4 },
    2: { pod: 1, points: 4 },
  }, {
    id: 'mock5',
    name: 'test5',
    points: 5,
    participated: true,
    dropped: false,
    1: { pod: 2, points: 5 },
    2: { pod: 2, points: 5 },
  }, {
    id: 'mock6',
    name: 'test6',
    points: 6,
    participated: true,
    dropped: false,
    2: { pod: 2, points: 6 },
  }],
}, {
  id: 'mock2',
  name: 'test2',
  category: 'cat0',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris vitae tellus nunc. Sed ullamcorper metus vitae nisi rhoncus placerat.',
  date: new Date('2018-01-01'),
  done: false,
  staging: false,
  pairingMethod: api.pairingMethods.RANDOM,
  pairingMethodInitial: api.pairingMethods.RANDOM,
  podSizeMinimum: 3,
  podSizeMaximum: 4,
  rounds: 1,
  players: [{
    id: 'mock0',
    name: 'test0',
    points: 0,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 0 },
  }, {
    id: 'mock1',
    name: 'test1',
    points: 1,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 1 },
  }, {
    id: 'mock2',
    name: 'test2',
    points: 2,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 2 },
  }, {
    id: 'mock3',
    name: 'test3',
    points: 3,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 3 },
  }],
}];

const mockRoute = {
  history: {
    push: jest.fn(to => to),
    replace: jest.fn(to => to),
  },
  match: {
    params: {},
  },
  matchWithInvalid: {
    params: {
      id: 'invalid',
    },
  },
  matchWithID0: {
    params: {
      id: 'mock0',
    },
  },
  matchWithID1: {
    params: {
      id: 'mock1',
    },
  },
};

let mockStore = {};
const mockStorage = {
  setItem: jest.fn((key, val) => Promise.resolve((mockStore[key] = val))),
  getItem: jest.fn(key => Promise.resolve(mockStore[key])),
  removeItem: jest.fn(key => { delete mockStore[key]; Promise.resolve(); }),
  clear: jest.fn(() => Promise.resolve((mockStore = {}))),
  keys: jest.fn(() => Promise.resolve(Object.keys(mockStore))),
  length: jest.fn(() => Promise.resolve(Object.keys(mockStore).length)),
};

let defaultStore = null;
function mockAPI(api, storage) {
  defaultStore = api.store;
  api.store = storage;
  mockStore = {
    settings: testSettings,
    players: testPlayers,
    tournaments: testTournaments,
  };
}
function demockAPI(api) {
  api.store = defaultStore;
  defaultStore = null;
}

export {
  testSettings,
  testPlayers,
  testTournaments,
  mockRoute,
  mockStore,
  mockStorage,
  mockAPI,
  demockAPI,
};
