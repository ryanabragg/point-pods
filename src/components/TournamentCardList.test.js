import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import TournamentCardList from './TournamentCardList';

import api from '../api';

import { mockStorage, mockAPI, demockAPI } from '../testData';

beforeEach(() => mockAPI(api, mockStorage));
afterEach(() => demockAPI(api));

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TournamentCardList />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('matches the prior snapshot', (done) => {
  const tree = renderer.create(<TournamentCardList />);
  expect(tree.toJSON()).toMatchSnapshot();
  setTimeout(() => {
    tree.update(<TournamentCardList />);
    expect(tree.toJSON()).toMatchSnapshot();
    done();
  }, 1000);
});

test('is a HOC-ed component', () => {
  const component = shallow(<TournamentCardList />);
  expect(component.type().prototype.constructor.name).toBe('APIComponent');
  expect(component.dive().type().prototype.constructor.name).toBe('TournamentCardList');
});

test('has control props: filterByDone, done', (done) => {
  const tree = renderer.create(<TournamentCardList />);
  setTimeout(() => {
    tree.update(<TournamentCardList filterByDone={true} />);
    expect(tree.toJSON()).toMatchSnapshot();
    tree.update(<TournamentCardList filterByDone={true} done={true} />);
    expect(tree.toJSON()).toMatchSnapshot();
    done();
  }, 1000);
});