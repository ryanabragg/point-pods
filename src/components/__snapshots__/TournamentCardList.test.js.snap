// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has control props: filterByDone, done 1`] = `
<div
  className="APIComponent-root-4"
>
  <div
    className="MuiPaper-root-8 MuiPaper-elevation1-11 MuiPaper-rounded-9 MuiCard-root-7 APIComponent-card-5"
  >
    <div
      className="MuiCardHeader-root-35"
    >
      <div
        className="MuiCardHeader-content-38"
      >
        <span
          className="MuiTypography-root-41 MuiTypography-headline-46 MuiCardHeader-title-39"
        >
          test0
        </span>
        <span
          className="MuiTypography-root-41 MuiTypography-body1-50 MuiTypography-colorTextSecondary-63 MuiCardHeader-subheader-40"
        >
          cat0
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root-65"
    >
      <p
        className="MuiTypography-root-41 MuiTypography-body1-50"
      >
        4 Total Players
      </p>
      <p
        className="MuiTypography-root-41 MuiTypography-body1-50"
      >
        4 Active Players
      </p>
    </div>
    <div
      className="MuiCardActions-root-66"
    >
      <a
        className="MuiButtonBase-root-92 MuiButton-root-68 MuiButton-text-70 MuiButton-flat-73 MuiButton-sizeSmall-89 APIComponent-right-6 MuiCardActions-action-67"
        href="/tournament/mock0"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-69"
        >
          Continue
        </span>
        <span
          className="MuiTouchRipple-root-95"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`has control props: filterByDone, done 2`] = `
<div
  className="APIComponent-root-4"
>
  <div
    className="MuiPaper-root-8 MuiPaper-elevation1-11 MuiPaper-rounded-9 MuiCard-root-7 APIComponent-card-5"
  >
    <div
      className="MuiCardHeader-root-35"
    >
      <div
        className="MuiCardHeader-content-38"
      >
        <span
          className="MuiTypography-root-41 MuiTypography-headline-46 MuiCardHeader-title-39"
        >
          test1
        </span>
        <span
          className="MuiTypography-root-41 MuiTypography-body1-50 MuiTypography-colorTextSecondary-63 MuiCardHeader-subheader-40"
        >
          cat1
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root-65"
    >
      <p
        className="MuiTypography-root-41 MuiTypography-body1-50"
      >
        7 Total Players
      </p>
      <p
        className="MuiTypography-root-41 MuiTypography-body1-50"
      >
        7 Active Players
      </p>
    </div>
    <div
      className="MuiCardActions-root-66"
    >
      <a
        className="MuiButtonBase-root-92 MuiButton-root-68 MuiButton-text-70 MuiButton-flat-73 MuiButton-sizeSmall-89 APIComponent-right-6 MuiCardActions-action-67"
        href="/tournament/mock1"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-69"
        >
          Details
        </span>
        <span
          className="MuiTouchRipple-root-95"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`matches the prior snapshot 1`] = `
<div
  className="APIComponent-root-4"
/>
`;

exports[`matches the prior snapshot 2`] = `
<div
  className="APIComponent-root-4"
>
  <div
    className="MuiPaper-root-8 MuiPaper-elevation1-11 MuiPaper-rounded-9 MuiCard-root-7 APIComponent-card-5"
  >
    <div
      className="MuiCardHeader-root-35"
    >
      <div
        className="MuiCardHeader-content-38"
      >
        <span
          className="MuiTypography-root-41 MuiTypography-headline-46 MuiCardHeader-title-39"
        >
          test0
        </span>
        <span
          className="MuiTypography-root-41 MuiTypography-body1-50 MuiTypography-colorTextSecondary-63 MuiCardHeader-subheader-40"
        >
          cat0
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root-65"
    >
      <p
        className="MuiTypography-root-41 MuiTypography-body1-50"
      >
        4 Total Players
      </p>
      <p
        className="MuiTypography-root-41 MuiTypography-body1-50"
      >
        4 Active Players
      </p>
    </div>
    <div
      className="MuiCardActions-root-66"
    >
      <a
        className="MuiButtonBase-root-92 MuiButton-root-68 MuiButton-text-70 MuiButton-flat-73 MuiButton-sizeSmall-89 APIComponent-right-6 MuiCardActions-action-67"
        href="/tournament/mock0"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-69"
        >
          Continue
        </span>
        <span
          className="MuiTouchRipple-root-95"
        />
      </a>
    </div>
  </div>
  <div
    className="MuiPaper-root-8 MuiPaper-elevation1-11 MuiPaper-rounded-9 MuiCard-root-7 APIComponent-card-5"
  >
    <div
      className="MuiCardHeader-root-35"
    >
      <div
        className="MuiCardHeader-content-38"
      >
        <span
          className="MuiTypography-root-41 MuiTypography-headline-46 MuiCardHeader-title-39"
        >
          test1
        </span>
        <span
          className="MuiTypography-root-41 MuiTypography-body1-50 MuiTypography-colorTextSecondary-63 MuiCardHeader-subheader-40"
        >
          cat1
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root-65"
    >
      <p
        className="MuiTypography-root-41 MuiTypography-body1-50"
      >
        7 Total Players
      </p>
      <p
        className="MuiTypography-root-41 MuiTypography-body1-50"
      >
        7 Active Players
      </p>
    </div>
    <div
      className="MuiCardActions-root-66"
    >
      <a
        className="MuiButtonBase-root-92 MuiButton-root-68 MuiButton-text-70 MuiButton-flat-73 MuiButton-sizeSmall-89 APIComponent-right-6 MuiCardActions-action-67"
        href="/tournament/mock1"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-69"
        >
          Details
        </span>
        <span
          className="MuiTouchRipple-root-95"
        />
      </a>
    </div>
  </div>
</div>
`;
